app/controllers/concerns/vendor_api/application_data_concerns.rb

```diff
@@ -0,0 +1,35 @@
+module VendorAPI
+  module ApplicationDataConcerns
+    PRELIMINARY_VERSION = '1.0'.freeze
+
+  private
+
+    def application_choice_id
+      params.permit(:application_id)[:application_id]
+    end
+
+    def application_choice
+      @application_choice ||= application_choices_visible_to_provider.find(application_choice_id)
+    end
+
+    def render_application
+      render json: application_json
+    end
+
+    def application_json
+      { data: ApplicationPresenter.new(version_number, application_choice).as_json }.to_json
```

Comment:
instead of:
```ruby
 %({"data":#{ApplicationPresenter.new(version_number, application_choice).serialized_json}})
```

Do we even need `serialized_json`?

Reply:
We need 2 methods of caching the application presenter output, and the differing outputs (JSON String or Ruby Hash) need distinct cache keys. It could well be done with one method. I think we needed a quick solution to a cache-overwrite bug at the time.

Reply:
reverted for now to use the cached JSON as this can be tackled on the ticket adding the meta.

Reply:
In terms of caching strings vs. Ruby hashes - as mentioned above I think it's fine to cache a Ruby Hash for the single application endpoint (though the index won't benefit from this), we need to be careful around cache keys as the String and Hash cache data need to be cached under different keys.